src/coord/src/coord.rs

```diff
@@ -359,6 +360,8 @@ pub struct Coordinator<S> {
     /// Handle to secret manager that can create and delete secrets from
     /// an arbitrary secret storage engine.
     secrets_controller: Box<dyn SecretsController>,
+    // Handle to secrets reader, wrapped in an arc so it can be used in async code like purification
+    secrets_reader: Arc<Box<dyn SecretsReader>>,
```

Comment:
`Arc` already causes a heap allocation, so the `Box` is redundant:

```suggestion
    secrets_reader: Arc<dyn SecretsReader>,
```